### PR TITLE
feat: [#6865] Provide two new invoke handlers, (message/fetchTask, message/submitAction) for Custom Feedback Loops

### PR DESCRIPTION
**Pull request by sw-joelmut**
Fixes # 6865
#minor

## Description
This PR introduces two new Teams handlers, submitAction and fetchTask, alongside with the respective feedback loop classes.

## Specific Changes
  - Added submitAction and fetchTask handlers for teams.
  - Added new classes for strong-type the new handlers' data.
  - Added new class for TeamsChannelData to enable feedback loops.
  - Added unit tests for the new handlers.

## Testing
The following image shows a teams sample working with the new submitAction handler, fetchTask couldn't be tested since it is only available in ring0.
img

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.Tests/Teams/TeamsActivityHandlerNotImplementedTests.cs [hunk 1]
```diff
@@ -688,6 +688,93 @@ void CaptureSend(Activity[] arg)
...
+
+            // Assert
+            Assert.NotNull(activitiesToSend);
+            Assert.Single(activitiesToSend);
+            Assert.IsType<InvokeResponse>(activitiesToSend[0].Value);
+            Assert.Equal(501, ((InvokeResponse)activitiesToSend[0].Value).Status);
+        }
+
+        [Fact]
+        public async Task TestMessageFetchTaskCustom()
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.Tests/Teams/TeamsActivityHandlerNotImplementedTests.cs:750]: This test is the same as the TestMessageFetchTaskDefault one.

File: tests/Microsoft.Bot.Builder.Tests/Teams/TeamsActivityHandlerTests.cs [hunk 2]
```diff
@@ -1512,6 +1513,108 @@ public async Task TestMessageDeleteActivityTeams_NoChannelData()
...
+
+            // Act
+            var bot = new TestActivityHandler();
+            await ((IBot)bot).OnTurnAsync(turnContext);
+
+            // Assert
+            Assert.NotNull(activitiesToSend);
+            Assert.Single(activitiesToSend);
+            Assert.IsType<InvokeResponse>(activitiesToSend[0].Value);
+            Assert.Equal(200, ((InvokeResponse)activitiesToSend[0].Value).Status);
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.Tests/Teams/TeamsActivityHandlerTests.cs:1615]: Let's add these assertions to differentiate this test from the previous one.
```suggestion
            // Assert
            Assert.Equal(2, bot.Record.Count());
            Assert.Equal("OnInvokeActivityAsync", bot.Record[0]);
            Assert.Equal("OnTeamsMessageFetchTaskAsync", bot.Record[1]);
            Assert.NotNull(activitiesToSend);
            Assert.Single(activitiesToSend);
            Assert.IsType<InvokeResponse>(activitiesToSend[0].Value);
            Assert.Equal(200, ((InvokeResponse)activitiesToSend[0].Value).Status);
```

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: tests/Microsoft.Bot.Builder.Tests/Teams/TeamsActivityHandlerTests.cs (hunk 2 above)
Comment (sw-joelmut) [reply tests/Microsoft.Bot.Builder.Tests/Teams/TeamsActivityHandlerTests.cs:1584]: Changed the two suggestions, thanks

File: libraries/Microsoft.Bot.Schema/Teams/FeedbackResponse.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        {
+            CustomInit();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FeedbackResponse"/> class.
+        /// </summary>
+        /// <param name="actionName">Unique identifier representing a team.</param>
+        /// <param name="actionValue">Unique identifier representing a team2.</param>
+        /// <param name="replyToId">Unique identifier representing a team3.</param>
```
[the remark this answers is not in the document]
Comment (sw-joelmut) [reply libraries/Microsoft.Bot.Schema/Teams/FeedbackResponse.cs]: Added missing doc, thanks

**Review by ceciliaavila — APPROVED** [review]

**Pull request closed without merging**